src/parser.rs

```diff
@@ -53,6 +55,20 @@ pub enum BasicParseErrorKind<'i> {
     QualifiedRuleInvalid,
 }
 
+impl<'i> BasicParseErrorKind<'i> {
+    fn description(&self) -> String {
+        match self {
+            BasicParseErrorKind::UnexpectedToken(token) => {
+                format!("Unexpected token: '{}'", token.description())
```

Comment:
I would just do: `format!("Unexpected token: {:?}", token)`, and remove `Token::description()`.